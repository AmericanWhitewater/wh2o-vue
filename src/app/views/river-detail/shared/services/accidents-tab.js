import { httpClient } from '@/app/global/services'

const fetchAccidentsData = data => {
  return httpClient
    .post('/graphql', {
      query: `
      query {
        reach(id: ${data}) {
          accidents(first: 20, page: 1) {
            data {
              accident_date
              water_level
              status
              factors {
                factor
              }
              injuries {
                injury
              }
              id
            }
          }
        }
      }`
    })
    .then(res => res.data)
}

export { fetchAccidentsData }
