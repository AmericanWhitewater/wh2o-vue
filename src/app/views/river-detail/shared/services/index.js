export * from './accidents-tab/accidents-tab'

export * from './bookmark-river/bookmark-river'

export * from './credits-tab/credits-tab'

export * from './flow-tab/flow-tab'

export * from './gallery-tab/gallery-tab'
/**
 * @todo refactor rapids
 */
export * from './main-tab/main-tab'
export * from './main-tab/comments'
export * from './main-tab/alerts'
export * from './rapids/rapids'

export * from './map-tab/map-tab'

export * from './weather-tab/weather-tab'
