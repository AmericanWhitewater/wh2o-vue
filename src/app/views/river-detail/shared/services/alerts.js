import { httpClient } from '@/app/global/services'

const fetchAlertsData = data => {
  return httpClient
    .post('/graphql', {
      query: `
      query {
        reach(id:${data}){
            posts(post_type: WARNING, first:10, page:1) {
                data {
                    id,
                    title,
                    detail,
                    post_date,
                    revision,
                    post_type,
                    user {
                        uname,
                        uid,
                        image {
                            uri{
                                thumb,
                                medium,
                                big
                            }
                        }
                    }
                }
            }
            }
        }
    
    `
    })
    .then(res => res.data)
}

const createAlert = () => {
  // does this need to be sent as JSON

  return httpClient.post('/graphql', {
    query: `
      mutation {
          postUpdate(
            id: "123456789", 
            post: { 
              post_type: COMMENT, 
              reach_id: "356", 
              title: "test", 
              user_id: "0", 
              detail: "test", 
              gauge_id: "sdf", 
              metric_id: "sdf", 
              post_date: "sdf", 
              reading: 1.5
            }
            ) {
            id
            uid
          }
        }`
  })
}

export { fetchAlertsData, createAlert }
