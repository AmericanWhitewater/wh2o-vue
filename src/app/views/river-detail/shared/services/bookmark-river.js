import { appLocalStorage, httpClient } from '@/app/global/services'
import { apiConstants } from '../config'
const merge_array = (array1, array2) => {
  var result_array = []
  var arr = array1.concat(array2)
  var len = arr.length
  var assoc = {}

  while (len--) {
    var item = arr[len]

    if (!assoc[item]) {
      result_array.unshift(item)
      assoc[item] = true
    }
  }

  return result_array
}

/**
 * store bookmarked rivers in user local storage
 * @param {string} riverId
 */
const bookmarkRiver = (riverId) => {
  const bookmarkRivers = appLocalStorage.getItem('wh2o-bookmarked-rivers')

  if (!bookmarkRivers) {
    appLocalStorage.setItem('wh2o-bookmarked-rivers', [riverId])
  } else {
    const updatedBookmarks = merge_array(bookmarkRivers, [riverId])
    appLocalStorage.setItem('wh2o-bookmarked-rivers', updatedBookmarks)
  }

  const data = appLocalStorage.getItem('wh2o-bookmarked-rivers')

  return data
}

const fetchBookmarksData = data => {
  const url = `${apiConstants.graphql}`
  return httpClient
    .post(url, {
      query: `
      query {
        reach(id:${data}) {
         river,
         section,
         class,
         id
      }
    }
    `
    })
    .then(res => res.data)
}
const fetchBookmarksGageData = data => {
  const url = `${apiConstants.graphql}`

  const gqlConfig = {
    query: `
      query {
        getGaugeInformationForReachID(id: ${data}) {
          gauges {
            gauge_max
            gauge_min
            class
            gauge_reading
          }
        }
      }`
  }

  return httpClient
    .post(url, gqlConfig)
    .then(res => res.data)
}

export { bookmarkRiver, fetchBookmarksData, fetchBookmarksGageData }
