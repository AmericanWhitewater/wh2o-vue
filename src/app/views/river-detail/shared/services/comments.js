import { httpClient } from '@/app/global/services'

import { apiConstants } from '../config'

const fetchCommentsData = data => {
  const url = `${apiConstants.graphql}`
  return httpClient
    .post(url, {
      query: `
        query Related{
            posts(reach_id: "${data}", post_types: COMMENT, page: 1, orderBy: {field: REVISION, order: DESC}, first: 10) {
              data {
                id
                title
                detail
                post_date
                revision
                post_type
                user {
                  uname
                  uid
                  image {
                    uri {
                      thumb
                      medium
                      big
                    }
                  }
                }
              }
            }
          }

    `
    })
    .then(res => res.data)
}

export { fetchCommentsData }
