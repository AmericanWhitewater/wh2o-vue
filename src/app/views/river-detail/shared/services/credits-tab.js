import { httpClient } from '@/app/global/services'

const fetchCreditsData = data => {
  return httpClient
    .post('/graphql', {
      query: `
      query {
        reach( id: ${data}) {
            revisions(first:100,page:0){data{id,revision,sk{uname},revision,revision_comment}}
        }
    }`
    })
    .then(res => res.data)
}

export { fetchCreditsData }
