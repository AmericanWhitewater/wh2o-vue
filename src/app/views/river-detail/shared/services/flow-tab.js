import { httpClient } from '@/app/global/services'
import { apiConstants } from '../config'
import axios from 'axios'
import { baseUrl } from '@/app/environment'

const fetchGaugeReadings = data => {
  /**
   * @todo remove proxy and setup local gage server
   * @todo format url to load from .env value
   * @temp get flow data from prod
   *
   */
  const url = `${baseUrl}api/gauge/${data.gauge_id}/flows/${data.metric_id}?from=${data.timeStart}&to=${data.timeEnd}&resolution=${data.resolution}`
  /**
   * @temp ideally keep axios in httpClient wrapper
   */
  return axios.get(url).then(res => res.data)
}

const fetchGaugeSourceInfo = id => {
  const url = apiConstants.gauge.base + id + apiConstants.gauge.source

  return httpClient.get(url).then(res => res.data)
}

const fetchGages = data => {
  return httpClient
    .post('/graphql', {
      query: `{
        getGaugeInformationForReachID(id: ${data}) {
          gauges {
              # rc
              # epoch
              gauge_reading
              gauge_metric
              gauge_comment
              range_comment
              class
              excluded
              rmin
              rmax
              gauge {
                  name
                  id
              }
              updated
              last_gauge_reading
              last_gauge_updated
              gauge_perfect
              adjusted_reach_class
            }
          }
        }`
    })
    .then(res => res.data)
}
const fetchGaugeMetrics = data => {
  return httpClient
    .post('/graphql', {
      query: `
      {
        getGaugeInformationForReachID(id: ${data}) {
            metrics {
              name
              unit
              format
              id
              shortkey
            }
          }
        }
    
    `
    })
    .then(res => res.data)
}

export { fetchGaugeReadings, fetchGaugeSourceInfo, fetchGaugeMetrics, fetchGages }
