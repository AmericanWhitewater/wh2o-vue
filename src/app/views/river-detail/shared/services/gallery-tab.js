import { httpClient } from '@/app/global/services'

const fetchGalleryData = data => {
  return httpClient
    .post('/graphql', {
      query: `
      query {
        posts(
          first: ${data.per_page},
          post_types: [JOURNAL,PHOTO_POST],
          reach_id: "${data.reach_id}", 
          page: ${data.page}, 
          orderBy: {field: REVISION, order: ASC}
          ) {
             data {
              id
              metric {
                name
                unit
              }
              gauge {
                id
                name
              }
              reading
              photos {
                image {
                  uri {
                    thumb
                    medium
                    big
                  }
                  file_size
                }
                id
                author
                caption
                description
                photo_date
                poi_name
                poi_id
                subject
              }
            }
            paginatorInfo {
              count
              perPage
              currentPage
              lastPage
              total
            }
        
          }  
        }`
    })
    .then(res => res.data)
}

export { fetchGalleryData }
