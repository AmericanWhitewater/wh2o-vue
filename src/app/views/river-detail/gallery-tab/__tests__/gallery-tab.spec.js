import GalleryTab from '@/app/views/river-detail/gallery-tab/gallery-tab.vue'
import { createWrapper } from '@/app/global/services'

const state = {
  userState: {
    userData: {
      data: null
    }
  },
  riverDetailState: {
    galleryData: {
      error: null,
      data: null,
      loading: null
    }
  }
}

const route = {
  params: {
    id: '357'
  }
}

describe('GalleryTab', () => {
  it('shows loading block when loading', async () => {
    state.riverDetailState.galleryData.loading = true

    const wrapper = createWrapper(GalleryTab, state, route)

    wrapper.setData({
      formattedData: null
    })

    expect(wrapper.find('.utility-block-loading').exists()).toBe(true)
    expect(wrapper.find('.utility-block-error').exists()).toBe(false)
    expect(wrapper.find('.utility-block-content').exists()).toBe(false)
  })

  it('shows error block when error', () => {
    state.riverDetailState.galleryData.loading = false
    state.riverDetailState.galleryData.error = true

    const wrapper = createWrapper(GalleryTab, state, route)

    expect(wrapper.find('.utility-block-loading').exists()).toBe(false)
    expect(wrapper.find('.utility-block-error').exists()).toBe(true)
    expect(wrapper.find('.utility-block-content').exists()).toBe(false)
  })

  it('shows no results block when error', () => {
    state.riverDetailState.galleryData.loading = false
    state.riverDetailState.galleryData.error = true
    state.riverDetailState.galleryData.data = []

    const wrapper = createWrapper(GalleryTab, state, route)

    expect(wrapper.find('.utility-block-loading').exists()).toBe(false)
    expect(wrapper.find('.utility-block-error').exists()).toBe(false)
    expect(wrapper.find('.utility-block-content').exists()).toBe(true)
  })
})
