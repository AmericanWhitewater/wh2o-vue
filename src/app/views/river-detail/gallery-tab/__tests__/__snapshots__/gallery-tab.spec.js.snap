// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryTab shows error block when error 1`] = `
<div id="utility-block" class="utility-block-error" style="height: 350px;">
  <!---->
  <!---->
  <div data-inline-loading="" role="alert" aria-live="assertive" class="bx--inline-loading" small="">
    <div class="bx--inline-loading__animation">
      <div class="bx--loading bx--loading--small" style="display: none;"><svg viewBox="-75 -75 150 150" class="bx--loading__svg">
          <circle cx="0" cy="0" r="26.8125" class="bx--loading__background"></circle>
          <circle cx="0" cy="0" r="26.8125" class="bx--loading__stroke"></circle>
        </svg></div><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" hidden="hidden" width="16" height="16" viewBox="0 0 16 16" aria-hidden="true" class="bx--inline-loading__checkmark-container" style="will-change: transform;">
        <path d="M8,1C4.1,1,1,4.1,1,8c0,3.9,3.1,7,7,7s7-3.1,7-7C15,4.1,11.9,1,8,1z M7,11L4.3,8.3l0.9-0.8L7,9.3l4-3.9l0.9,0.8L7,11z"></path>
        <path d="M7,11L4.3,8.3l0.9-0.8L7,9.3l4-3.9l0.9,0.8L7,11z" data-icon-path="inner-path" opacity="0"></path>
      </svg><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16" aria-hidden="true" class="bx--inline-loading--error" style="will-change: transform;">
        <path d="M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z M10.7,11.5L4.5,5.3l0.8-0.8l6.2,6.2L10.7,11.5z"></path>
        <path fill="none" d="M10.7,11.5L4.5,5.3l0.8-0.8l6.2,6.2L10.7,11.5z" data-icon-path="inner-path" opacity="0"></path>
      </svg>
    </div>
    <p class="bx--inline-loading__text">Loading data failed.</p>
  </div>
</div>
`;

exports[`GalleryTab shows loading block when loading 1`] = `
<div class="gallery-tab">
  <div class="bx--grid">
    <div class="bx--row">
      <div class="bx--col-sm-4 bx--col-md-8 bx--col-lg-10">
        <div id="utility-block" class="utility-block-loading" style="height: 350px;">
          <!---->
          <!---->
          <div data-inline-loading="" role="alert" aria-live="assertive" class="bx--inline-loading" small="">
            <div class="bx--inline-loading__animation">
              <div class="bx--loading bx--loading--small"><svg viewBox="-75 -75 150 150" class="bx--loading__svg">
                  <circle cx="0" cy="0" r="26.8125" class="bx--loading__background"></circle>
                  <circle cx="0" cy="0" r="26.8125" class="bx--loading__stroke"></circle>
                </svg></div><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" hidden="hidden" width="16" height="16" viewBox="0 0 16 16" aria-hidden="true" class="bx--inline-loading__checkmark-container" style="will-change: transform;">
                <path d="M8,1C4.1,1,1,4.1,1,8c0,3.9,3.1,7,7,7s7-3.1,7-7C15,4.1,11.9,1,8,1z M7,11L4.3,8.3l0.9-0.8L7,9.3l4-3.9l0.9,0.8L7,11z"></path>
                <path d="M7,11L4.3,8.3l0.9-0.8L7,9.3l4-3.9l0.9,0.8L7,11z" data-icon-path="inner-path" opacity="0"></path>
              </svg><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" hidden="hidden" width="16" height="16" viewBox="0 0 16 16" aria-hidden="true" class="bx--inline-loading--error" style="will-change: transform;">
                <path d="M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z M10.7,11.5L4.5,5.3l0.8-0.8l6.2,6.2L10.7,11.5z"></path>
                <path fill="none" d="M10.7,11.5L4.5,5.3l0.8-0.8l6.2,6.2L10.7,11.5z" data-icon-path="inner-path" opacity="0"></path>
              </svg>
            </div>
            <p class="bx--inline-loading__text">Loading data...</p>
          </div>
        </div>
      </div>
      <div class="bx--col-sm-4 bx--col-md-6 bx--col-lg-6 bx--col-max-5 bx--offset-max-1">
        <div class="gallery-upload-form">
          <hr>
          <h2>Upload Photos</h2>
          <div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--info">
            <div class="bx--inline-notification__details">
              <!---->
              <div class="bx--inline-notification__text-wrapper">
                <p class="bx--inline-notification__title">Login to Upload</p>
                <p class="bx--inline-notification__subtitle"></p>
              </div>
            </div>
            <!----><button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button"><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 32 32" aria-hidden="true" class="bx--inline-notification__close-icon" style="will-change: transform;">
                <path d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"></path>
              </svg></button>
          </div>
          <div class="cv-form-item bx--form-item cv-file-uploader"><strong class="bx--file--label"></strong>
            <p class="bx--label-description">Select the files you want to upload</p>
            <div data-file="" class="bx--file"><label for="uid-0698c96e-9fb0-4839-b61c-8bae9b5b57b0" role="button" tabindex="0" class="bx--file-browse-btn">
                <div data-file-drop-container="" class="bx--file__drop-container">Drag and drop files here or click to upload<input type="file" id="uid-0698c96e-9fb0-4839-b61c-8bae9b5b57b0" data-file-uploader="" data-target="[data-file-container]" accept="image/jpg,image/png" multiple="multiple" disabled="disabled" class="bx--file-input"></div>
              </label>
              <div data-file-container="" class="bx--file-container"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GalleryTab shows no results block when error 1`] = `
<div class="gallery-tab">
  <div class="bx--grid">
    <div class="bx--row">
      <div class="bx--col-sm-4 bx--col-md-8 bx--col-lg-10">
        <!---->
        <div id="utility-block" class="utility-block-content" style="height: 350px;">
          <h2 class="utility-block-title mb-spacing-md">No Media</h2>
          <p class="utility-block-text mb-spacing-md">if you have media for this reach, please share</p>
          <!---->
        </div>
      </div>
      <div class="bx--col-sm-4 bx--col-md-6 bx--col-lg-6 bx--col-max-5 bx--offset-max-1">
        <div class="gallery-upload-form">
          <hr>
          <h2>Upload Photos</h2>
          <div data-notification="" aria-live="polite" class="cv-inline-notification bx--inline-notification bx--inline-notification--info">
            <div class="bx--inline-notification__details">
              <!---->
              <div class="bx--inline-notification__text-wrapper">
                <p class="bx--inline-notification__title">Login to Upload</p>
                <p class="bx--inline-notification__subtitle"></p>
              </div>
            </div>
            <!----><button type="button" aria-label="Dismiss notification" data-notification-btn="" class="bx--inline-notification__close-button"><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 32 32" aria-hidden="true" class="bx--inline-notification__close-icon" style="will-change: transform;">
                <path d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"></path>
              </svg></button>
          </div>
          <div class="cv-form-item bx--form-item cv-file-uploader"><strong class="bx--file--label"></strong>
            <p class="bx--label-description">Select the files you want to upload</p>
            <div data-file="" class="bx--file"><label for="uid-bbae2cb5-7511-4b52-b21c-24a027691199" role="button" tabindex="0" class="bx--file-browse-btn">
                <div data-file-drop-container="" class="bx--file__drop-container">Drag and drop files here or click to upload<input type="file" id="uid-bbae2cb5-7511-4b52-b21c-24a027691199" data-file-uploader="" data-target="[data-file-container]" accept="image/jpg,image/png" multiple="multiple" disabled="disabled" class="bx--file-input"></div>
              </label>
              <div data-file-container="" class="bx--file-container"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
